### ISSUE TITLE
Switch between R versions in Rstudio?

### ISSUE THREAD

**Issue by shellytrigg** [RobertsLab/resources#925]
Has anyone actually had success getting Rstudio to switch between different versions of R? For instance to be able to run CRAN packages that are only available for earlier R versions?

I've been trying to follow [this guide](https://support.rstudio.com/hc/en-us/articles/200486138-Changing-R-versions-for-RStudio-desktop) for MacOS but with no luck. I recently upgraded my R to version 4.0 and my Rstudio to version 1.2.5042. I'm trying to use the package [MetStaT](https://github.com/cran/MetStaT) which is not available for R version 4.0.

**Comment by kubu4** [comment]
What happens when you do this (you have to set the path below to your desired R installation):

```export RSTUDIO_WHICH_R=/usr/local/bin/R```

Followed by starting R Studio from Terminal?

**Comment by sr320** [comment]
Try this Q on SAFS slack “help-with-code” channel.

**Comment by kubu4** [comment]
One other thing. If you look at [the CRAN web page for this package](https://cran.r-project.org/package=MetStaT), this is the message:

```
Package ‘MetStaT’ was removed from the CRAN repository.

Formerly available versions can be obtained from the archive.

Archived on 2019-12-14 at the request of the maintainer.

Please use the canonical form https://CRAN.R-project.org/package=MetStaT to link to this page.
```

Not sure this is a good sign (i.e. the people who wrote/maintained the package requested it to be removed from CRAN)...

**Comment by shellytrigg** [comment]
@kubu4 when I run:

```
export RSTUDIO_WHICH_R=/Library/Frameworks/R.framework/Versions/3.5/R
open -a Rstudio
```
Rstudio still starts up with 4.0 instead of 3.5. From Rstudio I see this:

> Sys.getenv('RSTUDIO_WHICH_R')
"/Library/Frameworks/R.framework/Versions/3.5/R"
> system("type R")
R is /usr/local/bin/R

@sr320 I haven't tried the renv suggestion so I'll look into that

**Comment by kubu4** [comment]
Unfortunately, you can't point the path directly to a framework. You would need to update this:

> Update the `R.framework/Versions/Current` directory alias directly using `ln -s`

I'd probably give the `Rswtich` software a shot (it's towards the bottom of the page):

https://mac.r-project.org/

**Comment by kubu4** [comment]
You could also try:

`install.packages('MetStaT'', repos='https://cran-archive.r-project.org')`

This tells R to use the archive mirror of CRAN (which is where MetStaT is currently stored on CRAN).

**Comment by kubu4** [comment]
Never mind. Still won't work using the CRAN archive....

**Comment by shellytrigg** [comment]
this seems promising: https://github.com/rstudio/renv/issues/254

**Comment by shellytrigg** [comment]
I got Rstuido to switch R versions using this app [https://rud.is/rswitch/](https://rud.is/rswitch/) suggested by Dan Ovando on the SAFS help with code channel. I was under the impression that this app just creates symlinks whichever R version you select. When I tried to do that manually I had no success. But it works with the app:

![Screen Shot 2020-05-07 at 10 49 29 AM](https://user-images.githubusercontent.com/4720221/81327646-7270cd00-9050-11ea-89fc-374f41fa8776.png)
